subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/metadata/CompilerMetadata.java

```diff
@@ -19,4 +19,5 @@
 import org.gradle.platform.base.internal.toolchain.ToolSearchResult;
 
 public interface CompilerMetadata extends ToolSearchResult {
+    String getVersion();
```

Comment:
What about calling this "Vendor" and treat that as something opaque that we pull from the compiler?  I think we don't care if this has version information.

This would also mean `CompilerVersion` would have a type, a vendor and a version string.  I haven't looked at what this would look like in build scans, but it might look better too. 

e.g., In the case where we have the same version of tools on two machines, but one is built for a different vendor, we would see `compilerVersion.vendor` is different between two builds (while `compilerVersion.version` and `compilerVersion.type` would be the same).

Reply:
I added `CompilerVersion.vendor`. This would also make sense for Java, the vendor being `Oracle` or `Azul Systems`. Maybe the type would then be something like `javac`, `errorprone` or `eclipse`.